.circleci/build_win.ps1

```diff
@@ -18,7 +18,7 @@ else {
 mkdir build
 cd build
 $boost_dir=(Resolve-Path $PSScriptRoot\..\deps\boost\lib\cmake\Boost-*)
-..\deps\cmake\bin\cmake -G "Visual Studio 16 2019" -DBoost_DIR="$boost_dir\" -DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded -DCMAKE_INSTALL_PREFIX="$PSScriptRoot\..\upload" ..
+..\deps\cmake\bin\cmake -G "Visual Studio 16 2019" -DBoost_DIR="$boost_dir\" -DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded -DCMAKE_INSTALL_PREFIX="$PSScriptRoot\..\upload" -DUSE_Z3=OFF ..
```

Comment:
I took a look at CI output and looks like we build with Z3 enabled but it's not installed so CMake does not find it. [`soltest.ps1`](https://github.com/ethereum/solidity/blob/develop/.circleci/soltest.ps1) runs tests with `--no-smt` anyway so it's not a problem.

Is this actually needed? I see nothing in your PR that would break because of it. We could add `-DUSE_Z3=OFF` here but it worked without it too.

Reply:
BTW, @leonardoalt Is SMT disabled on Windows because of non-determinism? I guess it's the reason why it's only macOS that's breaking your PRs :)

Reply:
we have another option, we can disable TESTS instead.

Reply:
b_win and b_win_release got failed, so I have to add this

Reply:
But what was the error? Maybe it's unrelated?

Reply:
> But what was the error? Maybe it's unrelated?

It failed because of my changes in cmake, the error was:
```
CMake Error at deps/cmake/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find Z3 (missing: Z3_LIBRARY Z3_INCLUDE_DIR) (Required is at
  least version "4.8.12")
Call Stack (most recent call first):
  deps/cmake/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindZ3.cmake:39 (find_package_handle_standard_args)
  CMakeLists.txt:68 (find_package)
```

Reply:
Ah, so it's exactly what I pointed out in https://github.com/ethereum/solidity/pull/11873#discussion_r705408383 :)

Without `REQUIRED` it should work.

Reply:
> I think it's disabled on windows because it's annoying to install (mainly).

Should I create an issue to add it eventually? Or are there other reasons why we might not want it? E.g. performance or that it might not add all that much value given we run on all other platforms?